wsi_deid/import_export.py

```diff
@@ -106,13 +107,38 @@ def getSchema():
 
     :returns: an object that can be passed to the jsonschema validator.
     """
-    # TODO: When we have a schema folder, if it exists and any items are in it
-    # that contain a single file that parses as valid json, then we need to
-    # load those files and combine them into a single schema in lieu of using
-    # this default SCHEMA_FILE_PATH
-    logger.info('Using %s (length %d) for schema' % (
-        SCHEMA_FILE_PATH, os.path.getsize(SCHEMA_FILE_PATH)))
-    return json.load(open(SCHEMA_FILE_PATH))
+    schemaFolderId = Setting().get(PluginSettings.WSI_DEID_SCHEMA_FOLDER)
+    mergedSchema = {'oneOf': []}
+
+    if schemaFolderId:
+        schemaFolder = Folder().load(schemaFolderId, force=True)
+        if schemaFolder is None:
+            return json.load(open(SCHEMA_FILE_PATH))
+        else:
+            max_files = 1000
+            fileList = list(itertools.islice(
+                Folder().fileList(schemaFolder, data=False),
+                max_files))
+            length = len(dict(fileList))
+
+            if length == 0:
+                return json.load(open(SCHEMA_FILE_PATH))
+            elif length == 1:
+                schemaFile = fileList[0][1]
+                print(schemaFile)
+                return json.load(File().open(schemaFile))
+
+            else:
+                for _filepath, file in fileList:
+                    try:
+                        if file['size'] < 2e+6:
+                            currentObject = json.load(File().open(file))
+                            mergedSchema['oneOf'].append(currentObject)
+                        else:
+                            pass
+                    except Exception:
+                        pass
+                return mergedSchema
```

Comment:
I think we probably *don't* want to recurse folders here, but only check items in the current folder.  This would let someone make a subfolder called something like "unused schemas" and manage schemas by moving them into that subfolder.  

In that case, it would probably be better to iterate over the items in the folder:
```
    if schemaFolderId:
        schemaFolder = Folder().load(schemaFolderId, force=True)
        if schemaFolder:
            max_items = 1000
            for item in Folder().childItems(schemaFolder, limit=max_items):
                # only consider items that contain exactly one file -- those that contain 0 or more than 1 probably aren't what we are looking for.
                if len(list(Item().childFiles(item, limit=2))) == 1:
                    file = next(Item().childFiles(item, limit=1))
                    if file['size'] < 2e+6:
                        try:
                            currentObject = json.load(File().open(file))
                            if 'properties' in currentObject:
                                mergedSchema['oneOf'].append(currentObject)
                        except Exception:
                            pass
    # Check the length of the collected files when we are all done, so that non-schema files don't change our choices.
    if not len(mergedSchema['oneOf']):
        return json.load(open(SCHEMA_FILE_PATH))
    if len(mergedSchema['oneOf']) == 1:
        return mergedSchema['oneOf'][0]
    return mergedSchema
```

Reply:
`if len(list(Item().childFiles(item, limit=2))) == 1:`
just curious as to why we limit to 2 here? Are there instances where the number of child files would be exorbitant?

Reply:
It really is just the principle of don't fetch more data than needed.  There could be lots of files (the most I've seen is some DICOM items that have a thousand-ish files), but each of those serializes out of the database to a few hundred bytes, and we really just want to check if there is exactly one, so there is no need to get them all.

Reply:
fixed in aaf5040ffda565dafaf99334ca6f82736d6aed6a